libs/sdk-ui-kit/src/@ui/UiMenu/itemUtils.ts

```diff
@@ -0,0 +1,138 @@
+// (C) 2025 GoodData Corporation
+
+import { IUiMenuInteractiveItem, IUiMenuItem } from "./types.js";
+
+export const findItem = <InteractiveItemData, StaticItemData>(
+    items: IUiMenuItem<InteractiveItemData, StaticItemData>[],
+    predicate: (item: IUiMenuItem<InteractiveItemData, StaticItemData>) => boolean,
+): IUiMenuItem<InteractiveItemData, StaticItemData> | undefined => {
+    for (const item of items) {
+        if (predicate(item)) {
+            return item;
+        }
+
+        if (item.type === "interactive" && item.subMenu) {
+            const foundItemInSubMenu = findItem(item.subMenu, predicate);
+
+            if (foundItemInSubMenu) {
+                return foundItemInSubMenu;
+            }
+        }
+    }
+    return undefined;
+};
+
+export const getItem = <InteractiveItemData, StaticItemData>(
+    items: IUiMenuItem<InteractiveItemData, StaticItemData>[],
+    id?: string,
+): IUiMenuItem<InteractiveItemData, StaticItemData> | undefined => {
+    return findItem(items, (item) => item.id === id);
+};
+
+export const findInteractiveItem = <InteractiveItemData, StaticItemData>(
+    items: IUiMenuItem<InteractiveItemData, StaticItemData>[],
+    predicate: (item: IUiMenuInteractiveItem<InteractiveItemData, StaticItemData>) => boolean,
+): IUiMenuInteractiveItem<InteractiveItemData, StaticItemData> | undefined => {
+    const foundItem = findItem(items, (item) => item.type === "interactive" && predicate(item));
+    return foundItem?.type === "interactive" ? foundItem : undefined;
+};
+
+export const getInteractiveItem = <InteractiveItemData, StaticItemData>(
+    items: IUiMenuItem<InteractiveItemData, StaticItemData>[],
+    id?: string,
+): IUiMenuInteractiveItem<InteractiveItemData, StaticItemData> | undefined => {
+    return findInteractiveItem(items, (item) => item.id === id);
+};
+
+export const getItemsInSubMenu = <InteractiveItemData, StaticItemData>(
+    items: IUiMenuItem<InteractiveItemData, StaticItemData>[],
+    id?: string,
+): IUiMenuItem<InteractiveItemData, StaticItemData>[] | undefined => {
+    return id == null ? items : findInteractiveItem(items, (item) => item.id === id)?.subMenu;
```

Comment:
this mixing of null and undefined is quite confusing. Why not to have std comparison id === undefined as expected for optional param? Does null have some special meaning, did author of code exchanged == and === on purpose or not?